### ISSUE TITLE
coral: update testing-library and related packages (hashtag1089)

### ISSUE THREAD

**Issue by vikramsubramanian** [Mayil-AI-Sandbox/klaw-f33g23#79]
We can update testing library and related packages to a newer version, but that requires some updates in the code. 

We should tackle that sometime, but it's not super important right now.

**Comment by vikramsubramanian** [comment]
I'm still thinking we might try convert from `jest` to `vitest`. Our tests are already pretty slow, and we don't have that many :o

**Comment by vikramsubramanian** [comment]
I'm 💯  for it! It's much faster 🥹 Give it a try! When we set up everything, we had some issues with typing (I wish I wrote that down in detail back then, but I'll check slack, not sure if I wrote with Samuli about it!) that might be solvable now. 

BUT - I feel a little bit queasy that it's still not in version `1.x` to be honest. I've no qualms using it in my side project stuff :D but we may feel different with Klaw? 🤔
